libs/s25main/mapGenerator/Textures.cpp

```diff
@@ -190,13 +198,21 @@ namespace rttr { namespace mapGenerator {
 
         ApplyCoastTexturing(coast, coastline);
 
-        auto mountainFoot = SelectPoints(
+        const auto mountainFoot = SelectPoints(
           [this](const MapPoint& pt) {
               return this->textureMap_.Any(pt, IsMinableMountain) && !this->textureMap_.All(pt, IsMountainOrSnowOrLava);
           },
           this->textures_.GetSize());
 
         ApplyMountainTransitions(mountainFoot);
+
+        const auto moutainWaterTransition = SelectPoints(
+          [this](const MapPoint& pt) {
```

Comment:
That was a lot of lambdas to follow ;)

Related: I've seen that you take the functor by instance, instead of by universal reference: https://github.com/Return-To-The-Roots/s25client/blob/7db186c6e58f72ca2ab2ecd080f325e264644a2c/libs/s25main/mapGenerator/Textures.h#L128 I'd usually suggest to use `T_Predicate&&` and forward as required or by const ref. Doesn't make a difference here as you are passing function pointers, but when you do benchmarking again you could check that.
You could also check, how the compiler would handle it, if you convert `SelectPoints` and `IsMinableMountain` & friends to inline functions instead of defining them in the cpp file. Maybe the compiler can inline all the lambdas/functions making this way more efficient. But of course should be measured first if this is worth it...

Just some ideas for the future I got while reviewing and wanted to mention them :)

Reply:
I only used those helper functions to avoid confusing stuff, e.g. by accidentally using two different ways to checking whether a texture is water or not. That was the original thought behind it. 

Just tried to move everything into header and put an "inline" upfront, but there was no real performance boost, unfortunately.